contracts/strategy/LiquidationStrategy.sol

```diff
@@ -230,113 +298,123 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
         bytes calldata data
     ) external onlyFixedSpreadLiquidationStrategy nonReentrant {
         LocalVars memory _vars;
-        (_vars.liquidatorAddress, _vars.tokenAdapter, _vars.router) = abi.decode(data, (address, IGenericTokenAdapter, IUniswapV2Router02));
-
-        // Retrieve collateral token
+        (_vars.liquidatorAddress, _vars.tokenAdapter, _vars.routerV2, _vars.routerV3, _vars.v2Ratio) = abi.decode(
+            data,
+            (address, IGenericTokenAdapter, address, address, uint256)
+        );
+        // Retrieve collateral token from CollateralTokenAdapter
         uint256 retrievedCollateralAmount = _retrieveCollateral(_vars.tokenAdapter, _collateralAmountToLiquidate);
-
+        // +1 to compensate for precision loss with division
         uint256 amountNeededToPayDebt = _debtValueToRepay.div(RAY) + 1;
 
-        (address[] memory path, uint256 dexAmountOut) = _computeMostProfitablePath(
-            _vars.router,
+        //dexAmountOut for rough calculation
+        (, uint256 dexAmountOut) = _computeMostProfitablePath(
+            IUniswapV2Router02(_vars.routerV2),
             _vars.tokenAdapter.collateralToken(),
             _collateralAmountToLiquidate
         );
 
-        /*
-         * LiquidationStrategy's condition quadrant
-         *
-         * +------------------+--------------------------------------+--------------------------------------+
-         * |                  | Sell WXDC to DEX                     | Not Sell WXDC to DEX                 |
-         * +------------------+--------------------------------------+--------------------------------------+
-         * | Allow Loss       | Sell WXDC and cover the loss         | X                                    |
-         * |                  | with FXD reserve in                  |                                      |
-         * |                  | LiquidationStrategy                  |                                      |
-         * +------------------+--------------------------------------+--------------------------------------+
-         * | Not Allow Loss   | Sell only if condition that          | Keep WXDC to LiquidationStrategy     |
-         * |                  | dexAmountOut > amountNeededToPayDebt | and deposit from LiquidationStrategy |
-         * |                  | and deposit FXD from WXDCSwap        |                                      |
-         * +------------------+--------------------------------------+--------------------------------------+
-         */
+        uint256 fathomStablecoinReceivedV2;
+        uint256 fathomStablecoinReceivedV3;
+
+        // Adjust v2Ratio if routerV3 is not set in uniswapV3Info
+        if (uniswapV3Info[_vars.routerV3].universalRouter == address(0)) {
+            _vars.v2Ratio = 10000; // Adjust to use V2 fully
+        }
 
         if (allowLoss == false) {
```

Comment:
looks like code duplication here and redundant if-else statements.
here we have: 
`if(allowLoss == false){
	if (dexAmountOut >= amountNeededToPayDebt) {
		// sell on DEX
	} else {
		// don't sell on DEX
	}
} else {
	// sell on DEX
}`

while it can be replaced with 
`if (allowLoss || dexAmountOut >= amountNeededToPayDebt) {
    // sell on dex
} else {
    // don't sell on dex
}`